### DISCUSSION TITLE
Bouncing collision into a corner and a moving platform (Also precise collision with a fast sprite)

### DISCUSSION THREAD

**Discussion by digisoul4** [MonoGame/MonoGame#8514] [Graphics]
Pong-like. Except...more of an Arkanoid-like? Whatever. Thing is, when the "ball" (its a square) collides with the corners of the stage with its own corner, it doesnt bounce. Also, when the "ball" goes too fast, it phases through the platforms its supposed to bounce on. What do? (The ball is meant to go a pixel faster after each bounce, so I need to avoid it from phasing through without making the sprites bigger or making it go slower)

**Comment by KelsamGames** [comment]
There are a bunch of ways to address this and if you google around for various 2D collision algorithms and 2D collision speed and such you'll find a lot to read and understand. But, presuming you're just trying to get your simple project working and are not too worried about quality: the method that is easiest, but worst performing, is to keep your existing collision code and simply set a maximum distance to move per collision check. I.e. if you need to move your ball 40 pixels but that will "phase through the wall" because it is only 10 pixels wide; then instead move 10 pixels, check for collision, move another 10, etc all in one update. Barely-thought-out-basic-algorithm: check if distance to move is greater than maximum (say, ball width/height); if so, instead move it the maximum then check collision and keep a var of how far its supposed to move and how far it has moved thus far, and keep looping until conditions are met so that it has moved the full distance.
